artifacts.go

```diff
@@ -100,6 +100,9 @@ type ArtifactConfig struct {
 	// Name is file or dir name to use for the artifact in the package.
 	// If empty, the file or dir name from the produced artifact will be used.
 	Name string `yaml:"name,omitempty" json:"name,omitempty"`
+	// Mode is the file mode to use for the artifact.
+	// Mode is represented as an fs.FileMode.
+	Mode fs.FileMode `yaml:"mode,omitempty" json:"mode,omitempty"`
```

Comment:
I think this should be called `permissions`

Reply:
I mean all the things.

```suggestion
	// Permissions is the file permissions to set on the artifact.
	// If not set, the default value will depend on the kind of artifact or the underlying artifact's already set permissions.
	Permissions fs.FileMode `yaml:"permissions,omitempty" json:"permissions,omitempty"`
```

Reply:
We use permissions everywhere except in `ArtifactDirConfig`, which we should probably change before 1.0